src/components/Router/useWorkflowSteps.tsx

```diff
@@ -0,0 +1,194 @@
+import { StepConfig } from '~types/steps'
+import { useCallback, useEffect, useRef, useState } from 'preact/hooks'
+
+import { formatStep } from '../../index'
+import { NarrowSdkOptions, UrlsConfig } from '~types/commons'
+import {
+  StepsLoadingStatus,
+  StepsHook,
+  CompleteStepValue,
+} from '~types/routers'
+import { poller, PollFunc, Engine } from '../WorkflowEngine'
+import type { WorkflowResponse } from '../WorkflowEngine/utils/WorkflowTypes'
+import useUserConsent from '~contexts/useUserConsent'
+
+type WorkflowStepsState = {
+  status: StepsLoadingStatus
+  steps: StepConfig[]
+  taskId: string | undefined
+  error: string | undefined
+}
+
+const defaultState: WorkflowStepsState = {
+  status: 'idle',
+  taskId: undefined,
+  error: undefined,
+  steps: [],
+}
+
+export const createWorkflowSteps = (
```

Comment:
Small suggestion here too, rename to `createWorkflowStepsHook` to better indicated what it does without needing to open the file. Also rename file to same